### PR TITLE
IE11 Banner

### PR DESCRIPTION
**Pull request by kylerwsm**
## Problem

Display to users via a banner that IE browsers may not work optimally on Go.gov.sg. This banner will only be shown to users on IE browsers.

## Solution

To check whether the browser is IE11, I am checking whether the browser supports `documentMode`, which is supported only in IE.

For consistency, the banner is implemented mobile-first even though there should not be any mobile phones running IE.

Reference to `documentMode`: https://www.w3schools.com/jsref/prop_doc_documentmode.asp

## Before & After Screenshots

**AFTER**:
image

### CODE REVIEW

File: src/client/components/BaseLayout/index.jsx [hunk 1]
```diff
@@ -40,10 +43,12 @@ const BaseLayout = ({
   children,
 }) => {
   const classes = useStyles()
+  const isIE = useIsIE()
   return (
     <>
       <CssBaseline />
       <Masthead />
+      {isIE && <BannerForIE />}
```
Comment (liangyuanruo) [inline src/client/components/BaseLayout/index.jsx:51]: We really only need to show this on the UserPage, can we move it there?

**Comment by kylerwsm** [comment]
Updated with proposed banner text. Banner is now only shown on IE users when they are on UserPage.

image

File: src/client/components/BaseLayout/index.jsx (hunk 1 above)
Comment (kylerwsm) [reply src/client/components/BaseLayout/index.jsx]: I have added an additional check to ensure that the banner only shows in UserPage for IE users.

**Comment by kylerwsm** [comment]
Resolved merge conflicts and cleared CI

**Pull request merged**